test/compare_results.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+from pytest import approx
+
+QUANTUM_ENERGY_IDX=4
+
+def find_start_end_block(lines, blockname):
+    """Identifies start and end of blocks"""
+    for i, line in enumerate(lines):
+        if "Start " + blockname in line:
+            start_idx = i
+        elif "End " + blockname in line:
+            end_idx = i
+    return start_idx, end_idx
+
+def extract_quantum_block(lines):
+    """Extracts only the quantum block from a chk file"""
+    idxs = find_start_end_block(lines, "Quantum")
+    temp = lines[idxs[0]+4:idxs[-1]]
+    return [line.split() for line in temp]
+
+def extract_quantum_energies(block):
+    return [float(line[QUANTUM_ENERGY_IDX]) for line in block]
+
+def read_chk_file(fname):
+    with open(fname, 'r') as fp:
+        lines = fp.readlines()
+        if lines:
+            lines  = filter( lambda x : x != '\n', lines ) # remove empty lines
+            lines  = filter( lambda x : '<-' not in x, lines ) # remove commented lines
+    return list(lines)
```

Comment:
Not necessary, but I would rewrite this with a list comprehension directly, mostly to avoid iterating over the lines twice:
```suggestion
    with open(fname, 'r') as fp:
        return [line for line in fp.readlines()
               if line != '\n' and '<-' not in line]
```

Can also use `if len(line.strip()) > 0` or something instead of `\n`, in case of whitespace weirdness.